src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -783,6 +821,30 @@ public FormValidation doCheckMachineType(@QueryParameter String value) {
             return FormValidation.ok();
         }
 
+        public FormValidation doCheckMaxRunDurationSeconds(
+                @QueryParameter String value, @QueryParameter(value = "provisioningType") String provisioningTypeIdx) {
```

Comment:
As per the method signature, this check should be invoked if either of `maxRunDurationSeconds` or radio button `provisioningType` is changed.

However it is only getting invoked if the `maxRunDurationSeconds` is invoked, despite `checkdependson` having `provisioningType`.

It seems the issue is with `provisioningType` being a radio button, if the jelly type was a `entry`, then the `doCheck` correctly gets invoked when either field is changed.

Reply:
Currently the `maxRunDuration` is only implemented in Spot and Standard provisioning types.

Much cleaner and better UX as well now, as the `preemptible` won't even show the option `maxRunDuration`